lib/path-handlers.js

```diff
@@ -47,6 +50,54 @@ function get_path(req, res, next) {
     });
 }
 
+
+function get_dir(req, res, next) {
+    var directories = [], juttles = [], relative_path, abs_path;
+
+
+    Promise.try(function() {
+        // massage the heck out of these paths
+        relative_path = req.query.path || '.';
+        relative_path = relative_path[0] === '/' ? relative_path.substring(1) : relative_path;
+        relative_path = path.normalize(relative_path);
+
+        abs_path = path.join(root_dir, relative_path);
+
+        if (abs_path.indexOf(root_dir) !== 0) {
+            throw errors.directoryAccessError(relative_path);
```

Comment:
Won't this always be true as you just did the join? If you're trying to make sure that someone didn't escape the root by having /../s in their path, I think you want to append the root first, then normalize, and then check to see if the root is still the beginning of the path.